### PR TITLE
[CFM-1248] Editing content author - fix autocomplete

### PR DESCRIPTION
**Pull request by anvmn**
https://issuetracker.amplexor.com/jira/browse/CFM-1248

### CODE REVIEW

File: project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module [hunk 1]
```diff
@@ -2527,8 +2527,11 @@ function c4m_og_add_content_author_options_to_power_users(&$form) {
     return;
   }
 
-  // Show authoring information tab to GAs and GO when editing group content.
-  $form['author']['#access'] = _c4m_features_og_members_is_power_user();
+  // Show authoring information tab to GAs and GO, when editing group content.
+  if (_c4m_features_og_members_is_power_user()) {
+    $form['author']['#access'] = TRUE;
+    $form['author']['date']['#access'] = FALSE;
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module]: add comment, why date access is `FALSE`

File: project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module [hunk 2]
```diff
@@ -2563,29 +2566,26 @@ function c4m_og_edit_author_autocomplete($string = '') {
...
+  $destination = explode('/', $destination['destination']);
+  $gid = $destination[2];
+
   // Return early if there is no group context.
   if (!$gid) {
     return;
   }
 
   $query = db_select('users', 'u');
-  $query
```
Comment (anvmn) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module]: @ordavidil 
Strange this query ever functioned properly.

File: project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module [hunk 3]
```diff
@@ -2563,29 +2566,28 @@ function c4m_og_edit_author_autocomplete($string = '') {
     return;
   }
 
-  $gid = c4m_og_current_group_id();
-  // Return early if there is no group context.
+  // Resolve group id from auto complte path, which is set to
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module]: => `complete`

File: project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module [hunk 4]
```diff
@@ -2563,29 +2566,28 @@ function c4m_og_edit_author_autocomplete($string = '') {
...
+  // 'edit-author/autocomplete/<$gid>'.
+  $destination = drupal_get_destination();
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module]: not sure about this one,  what are you trying to achieve ?

Comment (anvmn) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module]: Get the group id.
Previous implementation was relying on og_context(), which always returned NULL.

File: project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module [hunk 5]
```diff
@@ -2563,29 +2568,28 @@ function c4m_og_edit_author_autocomplete($string = '') {
     return;
   }
 
-  $gid = c4m_og_current_group_id();
-  // Return early if there is no group context.
+  // Resolve group id from auto complete path, which is set to
+  // 'edit-author/autocomplete/<$gid>'.
+  $destination = drupal_get_destination();
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module]: not sure about the `drupal_get_destination()`

Comment (anvmn) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module]: Tried to do this, but autocomplete is not as other menu items.
The parameter it gets is string to autocomplete.
`  $items['edit-author/autocomplete/%node'] = array(
    'page callback' => 'c4m_og_edit_author_autocomplete',`

`function c4m_og_edit_author_autocomplete($string = '')`

**Pull request merged**